src/classes/worker.ts

```
@@ -102,13 +103,23 @@ export class Worker<
       }
       this.timerManager = new TimerManager();
 
+      if (this.opts.process)
+        this.run().catch(error => {
+          this.running = false;
+          console.error(error);
+        });
+    }
+
+    this.on('error', err => console.error(err));
+  }
+
+  process(): void {
```

I think we can just make "run" public, so the error handling will be taken care by the caller. And inside the run() function we could just throw an exception if processFn is not defined (as it does not make sense to start the run loop if no process function is defined.

---

sure, let me refactor it a little bit